### ISSUE TITLE
HTML Block: Toggle syntax highlighting

### ISSUE THREAD

**Issue by mtias** [WordPress/gutenberg#5614]
Add a toggle to disable syntax highlighting in the inspector:

![image](https://user-images.githubusercontent.com/548849/37405981-5d02dcc6-2796-11e8-89dc-5cab8f1631cb.png)

## Considerations

Should this be a per-block settings or user setting? I feel if I want to disable syntax highlighting I'd appreciate it being disabled across the board.

cc @jasmussen

**Comment by jasmussen** [comment]
> I feel if I want to disable syntax highlighting I'd appreciate it being disabled across the board.

I think a per-block toggle would probably be frustrating, as you'd get tired of having to flip it for every new block you inserted. It seems a more global toggle would be better.

Should such a toggle live in the More menu, perhaps under the "Fix toolbar to block" toggle?

**Comment by mtias** [comment]
> Should such a toggle live in the More menu, perhaps under the "Fix toolbar to block" toggle?

I wouldn't mind it existing in the HTML Block inspector control (which is empty, and the first place I'd look) but being a global setting. (We can clarify that in the toggle.)

**Comment by karmatosed** [comment]
> Should such a toggle live in the More menu, perhaps under the "Fix toolbar to block" toggle?

I kind of like it in the more menu.

**Comment by bobbingwide** [comment]
Please can you explain this requirement. 

There is already a user level field labelled Syntax Highlighting which is used to control whether or not syntax is highlighted in the Customiser > Additional CSS and the Custom HTML Widget.
![issue 5614 - syntax highlighting](https://user-images.githubusercontent.com/2474435/38364730-ad6c1712-38d1-11e8-825f-28dd5dee6648.png)


- Is this the same thing? 
- If so, what are the requirements and what is the proposed solution?
- If not, why not? 
- What relation, if any, does this have to Andrea's comments on usability of the custom HTML block? https://github.com/WordPress/gutenberg/issues/5467#issuecomment-377161016

**Comment by westonruter** [comment]
I noticed that when I toggled the Syntax Highlighting option off that the Custom HTML block editor fails to render entirely.

**Comment by afercia** [comment]
Adding the accessibility label and 5.0 milestone as this is strictly related to #5467 that's already milestoned for 5.0.

**Comment by afercia** [comment]
Any chance to integrate this as a "global" option based on the existing user preference "Disable syntax highlighting when editing code"?

**Comment by afercia** [comment]
Discussed during today's accessibility bug-scrub and agreed ideally it should be based on the existing user option.

@aldavigdis also proposed to consider a "global" option for administrator who might want to disable it by default for all users, and users allowed to override the default.

**Comment by aldavigdis** [comment]
Just a soft suggestion. Putting things out in the ether.

**Comment by Mamaduka** [comment]
Closing this. The syntax highlighting has been removed from the HTML block.
